Use valid mnemonic by validating the checksum

NicolasDP opened the pull request.

vincenthz commented on test/GoldenTest.hs:
```diff
@@ -114,7 +114,7 @@ goldenHDWallet = group $ do
     runTest p (Mnemonic mw) pw ds (Root path) toSign =
         let -- 1. retrieve the seed
-            seed = cardanoSlSeed p mw
+            seed = fromMaybe undefined $ cardanoSlSeed p mw
```
probably make more sense to use an error with something that is easy to grep than `undefined`

vincenthz commented:
This is just missing a proper test with the official known BIP39 test vectors to make sure this is now solid

NicolasDP commented:
Yes, I have added the golden test from [BIP39 specs](https://github.com/bitcoin/bips/blob/master/bip-0039.mediawiki#test-vectors). Found a wee problem with the `entropyToWords` which is now fixed
